### ISSUE TITLE
DATA_OBJ_INFO.resc_id not being serialized correctly for pep_database_mod_data_obj_meta_post

### ISSUE THREAD

**Issue by tedgin** [irods/irods#6123]
- [x] main
- [x] 4-2-stable
---
### iRODS Version, OS and Version
iRODS 4.2.8, CentOS 7

### What did you try to do?
In a rule attached to the dynamic PEP `pep_database_mod_data_obj_meta_post(*INSTANCE, *CONTEXT, *OUT, *DATA_OBJ_INFO, *REG_PARAM)` I attempt to determine the resource Id number through `*DATA_OBJ_INFO.resc_id`.

### Expected behavior
I expect that the resource Id is a number.

### Observed behavior (including steps to reproduce, if applicable)
The value of `*DATA_OBJ_INFO.resc` is a single letter.

Here's an example.

```irods
pep_database_mod_data_obj_meta_post(*INSTANCE, *CONTEXT, *OUT, *DATA_OBJ_INFO, *REG_PARAM) {
  if (errorcode(*REG_PARAM.dataSize) == 0) {
    writeLine('serverLog', 'DATA_OBJ_INFO.resc_id = ' ++ DATA_OBJ_INFO.resc_id); 
  }
}
```

In the iRODS log, an entry like the following appeared.

```
Jan 21 15:00:10 pid:12000 NOTICE: writeLine: inString = DATA_OBJ_INFO.resc_id = q
```

**Comment by trel** [comment]
tl;dr; confirmed.

----
core.re
```
pep_database_mod_data_obj_meta_post(*INSTANCE, *CONTEXT, *OUT, *DATA_OBJ_INFO, *REG_PARAM){
    writeLine("serverLog",*DATA_OBJ_INFO);
}
```
running...
```
$ iput VERSION.json tedgin
```
gives this in the  rodsLog
```
Jan 21 17:10:21 pid:21733 NOTICE: writeLine: inString = KeyValue[29]:backup_resc_name=;checksum=;coll_id=0;data_access=;data_access_index=0;data_comments=;data_create=;data_expiry=;data_id=10078;data_map_id=0;data_mode=0;data_modify=01642803021;data_owner_name=;data_owner_zone=;data_size=224;data_type=generic;dest_resc_name=;flags=0;keyValPair_t=nullptr;logical_path=/tempZone/home/rods/tedgin;physical_path=/var/lib/irods/Vault/home/rods/tedgin;reg_uid=0;replica_number=0;replica_status=1;resc_hier=demoResc;resc_id=;sub_path=;version=;write_flag=0;
```

```
$ ils -L tedgin
  rods              0 demoResc          224 2022-01-21.17:10 & tedgin
        generic    /var/lib/irods/Vault/home/rods/tedgin
```

so...

`resc_id` is definitely not correct

but

`resc_hier` appears to be correct...

trying with a replication resource...

```
$ iput -R root VERSION.json tedgin2
$ ils -L tedgin2
  rods              0 root;repl;red          224 2022-01-21.17:12 & tedgin2
        generic    /tmp/redVault/home/rods/tedgin2
  rods              1 root;repl;green          224 2022-01-21.17:12 & tedgin2
        generic    /tmp/greenVault/home/rods/tedgin2
```

```
Jan 21 17:12:55 pid:21900 NOTICE: writeLine: inString = KeyValue[29]:backup_resc_name=;checksum=;coll_id=0;data_access=;data_access_index=0;data_comments=;data_create=;data_expiry=;data_id=10080;data_map_id=0;data_mode=0;data_modify=01642803175;data_owner_name=;data_owner_zone=;data_size=224;data_type=generic;dest_resc_name=;flags=0;keyValPair_t=nullptr;logical_path=/tempZone/home/rods/tedgin2;physical_path=/tmp/redVault/home/rods/tedgin2;reg_uid=0;replica_number=0;replica_status=1;resc_hier=root;repl;red;resc_id=$;sub_path=;version=;write_flag=0;
```

so confirmed... `resc_hier` is good/correct... `resc_id` is ... not.

we'll see if we can populate that more correctly.

**Comment by korydraughn** [comment]
needs checkboxes for main and 4-2-stable (this one should be checked).
